src/NewField.cpp

```diff
@@ -288,26 +366,34 @@ void NewFieldColumn::set_displayed_second(double second)
 
 double NewFieldColumn::calc_y_offset(double beat, double second)
 {
-	double note_beat= beat;
-	double curr_beat= m_curr_displayed_beat;
-	if(m_scroll_segments_enabled)
+	if(!m_in_defective_mode)
 	{
-		if(beat == m_curr_beat)
+		double note_beat= beat;
+		double curr_beat= m_curr_displayed_beat;
+		if(m_scroll_segments_enabled)
 		{
-			note_beat= m_curr_displayed_beat;
+			if(beat == m_curr_beat)
+			{
+				note_beat= m_curr_displayed_beat;
+			}
+			else
+			{
+				note_beat= m_timing_data->GetDisplayedBeat(static_cast<float>(note_beat));
+			}
 		}
-		else
+		mod_val_inputs input(note_beat, second, curr_beat, m_curr_second);
+		double ret= note_size * m_speed_mod.evaluate(input);
+		if(m_speed_segments_enabled)
 		{
-			note_beat= m_timing_data->GetDisplayedBeat(static_cast<float>(note_beat));
+			ret*= m_timing_data->GetDisplayedSpeedPercent(static_cast<float>(m_curr_beat), static_cast<float>(m_curr_second));
 		}
+		return ret;
 	}
-	mod_val_inputs input(note_beat, second, curr_beat, m_curr_second);
-	double ret= note_size * m_speed_mod.evaluate(input);
-	if(m_speed_segments_enabled)
+	else
 	{
-		ret*= m_timing_data->GetDisplayedSpeedPercent(static_cast<float>(m_curr_beat), static_cast<float>(m_curr_second));
+		return m_defective_mods->get_y_offset(beat, second, m_column);
 	}
-	return ret;
+	return 0.0;
```

Comment:
I put the unreachable return statement there to avoid a warning from a pedantic compiler.  Same goes for the 'f' on the ends of all the float constants.